### ISSUE TITLE
Json parsing missing multiple objects

### ISSUE THREAD

**Issue by zhewang2** [Kotlin/kotlinx.serialization#422]
We have integrated Kotlinx Serialization library in our spring boot application to replace Jackson. However, we found that the parsing function does not parsing specific fields from JSON object. 

For example:
"nativeTaxesAndFeePerStay": "49.08",
nativeTaxesAndFeePerStay is always parsed as null here
"nativeTaxesAndFeePerStay":null

Kotlin version: 1.3.21
kotlinx-serialization-runtime version: 0.11.0-1.3.30-eap-125

Here is the unit test class:
```
import kotlinx.serialization.json.Json
import org.junit.Test
import org.springframework.core.io.ClassPathResource

class RetailDetailsModelTest {

    @Test
    fun testHotelRoomRateModel() {
        val input = ClassPathResource("original_rates.json").file.readText()

        val data = Json.nonstrict.parse(HotelRoomRateModel.serializer(), input)

        println(Json.stringify(HotelRoomRateModel.serializer(), data))
    }

}
```

Here is the model class:
```
import kotlinx.serialization.Optional
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class HotelRoomRateModel( @Optional @SerialName( "rateIdentifier" ) val rateIdentifier: String? = null,
                               @Optional @SerialName( "programCode" ) val programCode: String? = null,
                               @Optional @SerialName( "targetProgramCode" ) val targetProgramCode: String? = null,
                               @Optional @SerialName( "ratePlanFilter" ) val ratePlanFilter: String? = null,
                               @Optional @SerialName( "planCode" ) val planCode: String? = null,
                               @Optional @SerialName( "roomCode" ) val roomCode: String? = null,
                               @Optional @SerialName( "gdsName" ) val gdsName: String? = null,
                               @Optional @SerialName( "gid" ) val gid: Int? = null,
                               @Optional @SerialName( "supplierId" ) val supplierId: Int? = null,
                               @Optional @SerialName( "rateTypeCode" ) val rateTypeCode: String? = null,
                               @Optional @SerialName( "rateAccessCode" ) val rateAccessCode: String? = null,
                               @Optional @SerialName( "rateCategoryType" ) val rateCategoryType: Int? = null,
                               @Optional @SerialName( "originalName" ) val originalName: String? = null,
                               @Optional @SerialName( "originalRoomRateDescription" ) val originalRoomRateDescription: String? = null,
                               @Optional @SerialName( "maxOccupancy" ) val maxOccupancy: Int? = null,
                               @Optional @SerialName( "pricedOccupancy" ) val pricedOccupancy: Int? = null,
                               @Optional @SerialName( "maxChildrenFree" ) val maxChildrenFree: Int? = null,
                               @Optional @SerialName( "maxChildrenFreeAge" ) val maxChildrenFreeAge: Int? = null,
                               @Optional @SerialName( "totalRoomsRemaining" ) val totalRoomsRemaining: Int? = null,
                               @Optional @SerialName( "exchangeRate" ) val exchangeRate: String? = null,
                               @Optional @SerialName( "nativeExchangeRate" ) val nativeExchangeRate: String? = null,
                               @Optional @SerialName( "averageNightlyRate" ) val averageNightlyRate: String? = null,
                               @Optional @SerialName( "nativeAverageNightlyRate" ) val nativeAverageNightlyRate: String? = null,
                               @Optional @SerialName( "nativeNightlyRates" ) val nativeNightlyRates: List<String>? = null,
                               @Optional @SerialName( "nightlyRates" ) val nightlyRates: List<String>? = null,
                               @Optional @SerialName( "baseRate" ) val baseRate: String? = null,
                               @Optional @SerialName( "nativeBaseRate" ) val nativeBaseRate: String? = null,
                               @Optional @SerialName( "marginAmount" ) val marginAmount: String? = null,
                               @Optional @SerialName( "nativeMarginAmount" ) val nativeMarginAmount: String? = null,
                               @Optional @SerialName( "taxRate" ) val taxRate: Double? = null,
                               @Optional @SerialName( "processingFeePerStay" ) val processingFeePerStay: String? = null,
                               @Optional @SerialName( "nativeProcessingFeePerStay" ) val nativeProcessingFeePerStay: String? = null,
                               @Optional @SerialName( "originalProcessingFeePRPN" ) val originalProcessingFeePRPN: String? = null,
                               @Optional @SerialName( "adjustedProcessingFeePRPN" ) val adjustedProcessingFeePRPN: String? = null,
                               @Optional @SerialName( "taxesAndFeePerStay" ) val taxesAndFeePerStay: String? = null,
                               @Optional @SerialName( "nativeTaxesAndFeePerStay" ) val nativeTaxesAndFeePerStay: Boolean? = null,
                               @Optional @SerialName( "hotelTaxesPerStay" ) val hotelTaxesPerStay: String? = null,
                               @Optional @SerialName( "nativeHotelTaxesPerStay" ) val nativeHotelTaxesPerStay: String? = null,
                               @Optional @SerialName( "totalPriceExcludingTaxesAndFeePerStay" ) val totalPriceExcludingTaxesAndFeePerStay: String? = null,
                               @Optional @SerialName( "nativeTotalPriceExcludingTaxesAndFeePerStay" ) val nativeTotalPriceExcludingTaxesAndFeePerStay: String? = null,
                               @Optional @SerialName( "totalPriceIncludingTaxesAndFeePerStay" ) val totalPriceIncludingTaxesAndFeePerStay: String? = null,
                               @Optional @SerialName( "nativeTotalPriceIncludingTaxesAndFeePerStay" ) val nativeTotalPriceIncludingTaxesAndFeePerStay: String? = null,
                               @Optional @SerialName( "grandTotal" ) val grandTotal: String? = null,
                               @Optional @SerialName( "nativeGrandTotal" ) val nativeGrandTotal: String? = null,
                               @Optional @SerialName( "currencyCode" ) val currencyCode: String? = null,
                               @Optional @SerialName( "currencySymbol" ) val currencySymbol: String? = null,
                               @Optional @SerialName( "nativeCurrencyCode" ) val nativeCurrencyCode: String? = null,
                               @Optional @SerialName( "nativeCurrencySymbol" ) val nativeCurrencySymbol: String? = null,
                               @Optional @SerialName( "refundPolicy" ) val refundPolicy: String? = null,
                               @Optional @SerialName( "payWhenYouStayFlag" ) val payWhenYouStayFlag: Boolean? = null,
                               @Optional @SerialName( "roomsLeft" ) val roomsLeft: Int? = null,
                               @Optional @SerialName( "rmiScore" ) val rmiScore: Double? = null,
                               @Optional @SerialName( "rateLevelPolicies" ) val rateLevelPolicies:  Map<String, String>? = null,
                               @Optional @SerialName( "paymentOptions" ) val paymentOptions: Set<String>? = null,
                               @Optional @SerialName( "checkInPaymentOptions" ) val checkInPaymentOptions: Set<String>? = null,
                               @Optional @SerialName( "totalTaxes" ) val totalTaxes: List<String>? = null,
                               @Optional @SerialName( "totalTaxesUSD" ) val totalTaxesUSD: List<String>? = null,
                               @Optional @SerialName( "processingFees" ) val processingFees: List<String>? = null,
                               @Optional @SerialName( "processingFeesUSD" ) val processingFeesUSD: List<String>? = null,
                               @Optional @SerialName( "totalTaxesAndFees" ) val totalTaxesAndFees: List<String>? = null,
                               @Optional @SerialName( "totalTaxesAndFeesUSD" ) val totalTaxesAndFeesUSD: List<String>? = null,
                               @Optional @SerialName( "processingFeeId" ) val processingFeeId: Long? = null,
                               @Optional @SerialName( "programName" ) val programName: String? = null,
                               @Optional @SerialName( "strikeThroughPrice" ) val strikeThroughPrice: String? = null,
                               @Optional @SerialName( "savingsPct" ) val savingsPct: String? = null,
                               @Optional @SerialName( "merchandisingFlag" ) val merchandisingFlag: Boolean? = null,
                               @Optional @SerialName( "displayName" ) val displayName: String? = null,
                               @Optional @SerialName( "programCategoryName" ) val programCategoryName: String? = null,
                               @Optional @SerialName( "channelName" ) val channelName: String? = null,
                               @Optional @SerialName( "rateTypeGroupName" ) val rateTypeGroupName: String? = null,
                               @Optional @SerialName( "merchandisingText" ) val merchandisingText: String? = null,
                               @Optional @SerialName( "partialUnlock" ) val partialUnlock: Boolean? = null,
                               @Optional @SerialName( "hotelEntityRuleId" ) val hotelEntityRuleId: Long? = null,
                               @Optional @SerialName( "ccRequired" ) val ccRequired: Boolean? = null,
                               @Optional @SerialName( "cvcRequired" ) val cvcRequired: Boolean? = null,
                               @Optional @SerialName( "targetBooking" ) val targetBooking: Boolean? = null,
                               @Optional @SerialName( "ratePolicies" ) val ratePolicies:  Map<String, String>? = null,
                               @Optional @SerialName( "suggestedNumOfRooms" ) val suggestedNumOfRooms: Int? = null,
                               @Optional @SerialName( "markupAdjusted" ) val markupAdjusted: Boolean? = null,
                               @Optional @SerialName( "masterRoomId" ) val masterRoomId: String? = null,
                               @Optional @SerialName( "optDbgInfo" ) val optDbgInfo: String? = null,
                               @Optional @SerialName( "billingAddressRequired" ) val billingAddressRequired: Boolean? = null,
                               @Optional @SerialName( "markupAdjustmentCrxId" ) val markupAdjustmentCrxId: Long? = null,
                               @Optional @SerialName( "hotelProgramCategoryId" ) val hotelProgramCategoryId: Long? = null,
                               @Optional @SerialName( "dynamicMarkupControllerId" ) val dynamicMarkupControllerId: Long? = null,
                               @Optional @SerialName( "freeCancellation" ) val freeCancellation: Boolean? = null,
                               @Optional @SerialName( "taxPrototypeIds" ) val taxPrototypeIds: List<String>? = null,
                               @Optional @SerialName( "couponApplicable" ) val isCouponApplicable: Boolean? = null )
```

Here is the JSON
```
{
  "rateIdentifier": "8A2ABF6F80343579EA67F58AAF6D2AE686A0D76C16A0EAA6F49F2D5CBB63C96E4ABA626275F30B7AA963FC682DE55B217EA3048807BEE935",
  "programCode": "AGENCY",
  "targetProgramCode": "AGENCY",
  "planCode": "BKG",
  "roomCode": "2529411_119278381_0_2_0",
  "gdsName": "BOOKING",
  "gid": 2826,
  "rateTypeCode": "AGENCY",
  "rateCategoryType": 8,
  "originalName": "Double Bed + Sofabed - Deluxe Room",
  "originalRoomRateDescription": "Deluxe Double Room with Sofa Bed - Non-refundable - This room features a sitting area with a sofa bed and floor-to-ceiling windows. Additional facilities include soundproofing, a flat-screen TV and an iPod dock.\n\nNot all rates include breakfast. If you book a breakfast inclusive rate, breakfast is provided for 2 adults and up to 2 children under 12 years old.",
  "maxOccupancy": 6,
  "pricedOccupancy": 4,
  "totalRoomsRemaining": 5,
  "exchangeRate": "1",
  "averageNightlyRate": "309.00",
  "nativeAverageNightlyRate": "309.00",
  "nativeNightlyRates": [
    "309.00"
  ],
  "nightlyRates": [
    "309.00"
  ],
  "taxRate": 0,
  "processingFeePerStay": "0.00",
  "nativeProcessingFeePerStay": "0.00",
  "originalProcessingFeePRPN": "0.00",
  "adjustedProcessingFeePRPN": "0.00",
  "taxesAndFeePerStay": "49.08",
  "nativeTaxesAndFeePerStay": "49.08",
  "totalPriceExcludingTaxesAndFeePerStay": "309.00",
  "nativeTotalPriceExcludingTaxesAndFeePerStay": "309.00",
  "totalPriceIncludingTaxesAndFeePerStay": "358.08",
  "nativeTotalPriceIncludingTaxesAndFeePerStay": "358.08",
  "grandTotal": "392.51",
  "nativeGrandTotal": "392.51",
  "currencyCode": "USD",
  "currencySymbol": "$",
  "nativeCurrencyCode": "USD",
  "nativeCurrencySymbol": "$",
  "cancellationPolicy": {
    "text": " You can cancel free of charge until 3 days before arrival. You'll be charged the total price if you cancel in the 3 days before arrival.",
    "cancelPolicyCategory": "NON_REFUNDABLE",
    "freeRefundableFlag": false
  },
  "payWhenYouStayFlag": true,
  "roomsLeft": 5,
  "rmiScore": 1,
  "mandatoryPropertyFees": {
    "feeAmountPerRoom": "34.43",
    "currencyCode": "USD",
    "feeAmountPerRoomNative": "34.43",
    "nativeCurrencyCode": "USD",
    "totalFeeAmount": "34.43",
    "totalFeeAmountNative": "34.43",
    "mandatoryFeeSummaries": [
      {
        "value": 34.43,
        "currency": "USD",
        "type": "AMOUNT",
        "units": "PER_ROOM_NIGHT"
      }
    ],
    "mandatoryFeeDetails": [
      {
        "id": 0,
        "name": "resort fee",
        "value": 34.43,
        "amountPerRoomInNativeCurrency": "34.43",
        "amountPerRoomInRequestedCurrency": "34.43",
        "measureType": "A",
        "calculationCode": "PRN",
        "calculationDescription": "per room per night"
      }
    ]
  },
  "roomRateFeatures": {
    "beddingList": [
      {
        "bedType": "DOUBLE",
        "sofa": false
      },
      {
        "sofaType": "SOFABED",
        "sofa": true
      }
    ],
    "multiBedsLogicalOperators": [
      "+"
    ],
    "roomType": "DELUXE",
    "amenities": [
      "FREE WIFI"
    ],
    "version": 26
  },
  "rateLevelPolicies": {
    "POLICY_HOTEL_PARKING": "Public parking is available at a location nearby (reservation is not possible) and costs  USD 90 per  day. ",
    "POLICY_PREPAY": "Your credit card will be used to guarantee the reservation. The hotel will charge you directly for room and tax. No prepayment is needed. The hotel reserves the right to authorize your credit card prior to check-in.\n\nIf you dont check-in to the hotel on the first day of your reservation and you do not alert the hotel in advance, the hotel reserves the right to cancel your reservation and may charge you a penalty.",
    "POLICY_HOTEL_PETS": "Pets are not allowed. ",
    "POLICY_PHOTO": "The reservation holder must present a valid photo ID and credit card at check-in. The credit card is required for <b>the mandatory fee listed above as well as</b> any additional hotel incidental charges such as parking, phone calls or minibar charges which are not included in the room rate.",
    "POLICY_HOTEL_GROUPS": "When booking more than 9 rooms, different policies and additional supplements may apply. ",
    "POLICY_HOTEL_INTERNET": "WiFi is available in the hotel rooms and costs  USD  16.95 per  24 hours. ",
    "POLICY_CHILDREN": "All children are welcome. One additional child under 3 years stays free of charge in a crib.\nThere is no capacity for extra beds in the room.\n",
    "IMPORTANT_INFORMATION": "Guests are required to show a photo ID and credit card upon check-in. Please note that all Special Requests are subject to availability and additional charges may apply.\nThe daily destination fee includes the following:\nUSD 35 food credit\nEnhanced Internet.\n1 cycle class",
    "POLICY_RATE_DESC": "<b>Rate Description: </b>Special Rate",
    "POLICY_PREAUTHORIZE": "The property reserves the right to pre-authorize credit cards prior to arrival.",
    "POLICY_HOTEL_OCCUPANCY": "All rooms are booked for double occupancy (i.e. 2 adults) and accommodations for more than two adults are not guaranteed.",
    "POLICY_HOTEL_EXTRACHARGES": "USD 49.08 tax per room is excluded from the rate. USD 34.43 hotel mandatory fees per room are excluded from the rate."
  },
  "paymentOptions": [
    "VI",
    "MC",
    "AX"
  ],
  "checkInPaymentOptions": [
    "American Express",
    "JCB",
    "Visa",
    "Discover",
    "Diners Club",
    "Mastercard"
  ],
  "ccRequired": true,
  "cvcRequired": false,
  "targetBooking": true,
  "billingAddressRequired": false,
  "couponApplicable": false
}
```

**Comment by zhewang2** [comment]
Btw, how to debug visitor code?

**Comment by fvasco** [comment]
See https://github.com/Kotlin/kotlinx.serialization/issues/377

**Comment by zhewang2** [comment]
@fvasco thank you so much for sharing the related issue.
Do we know when the fix will be available?

**Comment by zhewang2** [comment]
Duplicate with #377

**Comment by fvasco** [comment]
[Soon](https://github.com/Kotlin/kotlinx.serialization/issues/377#issuecomment-463973773)

**Comment by zhewang2** [comment]
Fixed by release 1.3.30.

**Comment by zhewang2** [comment]
> [Soon](https://github.com/Kotlin/kotlinx.serialization/issues/377#issuecomment-463973773)

Thank you for replying!
